### PR TITLE
Topic/renovatebot

### PR DESCRIPTION
**Pull request by ohdearaugustin**
- [x] read the [CONTRIBUTING guidelines](README.md#user-content-contributing)
- [x] raised a GitHub issue or discussed it on the projects chat beforehand
- [] added unit tests
- [] added integration tests
- [] updated documentation if needed
- [] updated CHANGELOG.md

This will implement #323. 
To make the renovatebot work, we would need to create a github app. (I decided to implement it with an app, as the permissions can be set more specifically)

In the developer setting create a new app.
1. Set GitHub App name (eg.renovatebot-headscale, or something like that)
2. Set Homepage URL to https://github.com/juanfont/headscale
3. Set permissions:
- Administration Read-only
- Contents Read & write
- Issues Read & write
- Metadata Read-only
- Pull requests Read & write
- Workflows Read & write
- Everything else leave on No access
4. Install App "Only select repositories" --> juanfont/headscale
5. Generate Private Key (this will let you download a zip with a cert inside)
6. Add to the Secrets Actions of the repository two new secrets
7. *RENOVATEBOT_APP_ID* with the content App ID
8. *RENOVATEBOT_SECRET* with the private key encoded in base64 (cat <APPNAME>.<CURRENT_DATE>.private-key.pem | base64 -w 0 && echo
9. Merge PR

After the setup of renovatebot it run it manually for the first time. This will result in some PRs to update some github actions. After that the renovatebot will look automatically on the 5th and 20th of each month for updates. Otherwise it will spam a lot of PRs for update. This can be reconfigured if needed.

### CODE REVIEW

File: .github/renovate.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  "gitAuthor": "Renovate Bot <bot@renovateapp.com>",
+  "branchPrefix": "renovateaction/",
+  "onboarding": false,
+  "extends": ["config:base", ":rebaseStalePrs"],
+  "ignorePresets": [":prHourlyLimit2"],
+  "enabledManagers": ["dockerfile", "gomod", "github-actions","regex" ],
+  "includeForks": true,
+  "repositories": ["juanfont/headscale"],
+  "platform": "github",
+  "packageRules": [
```
Comment (kradalby) [inline .github/renovate.json:13]: Can we make it also update the go version in github actions?

**Review by kradalby — APPROVED** [review]
I'll approve this, but @juanfont has to do the steps

File: .github/renovate.json (hunk 1 above)
Comment (ohdearaugustin) [reply .github/renovate.json:13]: The regexManagers will update the go-verisons of the github actions, see line 26.

**Comment by juanfont** [comment]
Great work!

**Pull request merged**